File: src/compile_lossfactors.R
rm(list = ls())

library(tidyverse)
library(vroom)

dir_output <- "Y:/gem/10 Research/10 Projects/GREENFIN/70 Equity VaR Tipping Points/META_results_Aug2024"

mc_samplesize <- 10000

dir_to_compile <- list.files(dir_output) %>% str_subset(paste0("^n", mc_samplesize))

start <- Sys.time()
dir_selected <- dir_to_compile[2]
variable_selected <- "lossfactor"

if(!variable_selected %in% c("T_AT", "SLR", "lossfactor")) stop("Issue")

paths_selected <- file.path(dir_output, dir_selected,
                       ifelse(variable_selected == "lossfactor", "lossfactors_conspc", variable_selected),
                       paste0("variable_selected_mc", 1:mc_samplesize, ".csv"))

if(length(paths_selected) != mc_samplesize) stop("Unexpected # of files")

if(mean(map_lgl(paths_selected, file.exists)) != 1) stop("Not all MC files exist")


df_out <- vroom::vroom(paths_selected, show_col_types = F, col_types = "d")
write_csv(df_out, file.path(dir_output, dir_selected, paste0(variable_selected, ".csv")))
end <- Sys.time()

end - start

File: src/compile_lossfactors.jl
using CSV
using DataFrames

dir_output = "Y:/gem/10 Research/10 Projects/GREENFIN/70 Equity VaR Tipping Points/META_results_Aug2024"

subfolders = readdir(dir_output) |> filter(f -> occursin(r"n10000", f) && !occursin(r"n10000\.csv$", f))

for subfolder_jj in subfolders
    
    print(subfolder_jj)

    # Get a list of all CSV files in the directory
    files_slr = readdir(string(dir_output, "/", subfolder_jj, "/SLR")) |> filter(f -> occursin(r"SLR_mc\d+\.csv$", f) )

    # Initialize an empty DataFrame to hold the combined data
    combined_df = DataFrame()

    # Loop through each file, read it, and append it to the combined DataFrame
    for file_jj in files_slr
        df = CSV.read(string(dir_output, "/", subfolder_jj, "/SLR/", file_jj), DataFrame)
        append!(combined_df, df)
    end

    # Save the combined DataFrame as a CSV file
    CSV.write(string(dir_output, "/", subfolder_jj, "/SLR.csv"), combined_df)
end
